### PR TITLE
Use Gradle's lazy APIs when possible

### PR DESCRIPTION
**Pull request by eygraber**

### CODE REVIEW

File: build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts [hunk 1]
```diff
@@ -19,26 +20,12 @@ kotlin {
...
-        for (sourceSet in nativeTargets) {
-            getByName("${sourceSet}Main") {
-                dependsOn(nativeMain)
-            }
-            getByName("${sourceSet}Test") {
-                dependsOn(nativeTest)
-            }
-        }
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    targetHierarchy.default()
```
Comment (eygraber) [inline build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts:24]: This is a really cool feature that sets up the source sets the way you'd expect them to be (e.g. `native` is set up), and it is extensible.

It is experimental, so I can remove it if you want.

Comment (evant) [reply build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts:24]: kotlin multiplatform is experimental anyway so it's fine. Maybe try a quick `./gradlew publichToMavenLocal` to make sure there isn't any issues with publishing with the new layout.

**Review by evant — COMMENTED** [review]
registering, existing, gradle is running out of words aren't they? lol

**Comment by evant** [comment]
hey just an fyi, when you stack pr's it's nicer to have the stacked pr point to the original as the base branch instead of main, it makes the diff easier to read. github will automatically retarget when the original one gets merged.

**Comment by eygraber** [comment]
I think the issue with not pointing at main is that it keeps the PR in my fork, and I don't think it's possible to re-point it to this repo after the target gets merged.

File: build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts (hunk 1 above)
Comment (eygraber) [reply build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts:24]: LGTM although I'm building on Linux so only the linuxX64 target gets built for native.

**Comment by evant** [comment]
bah yeah you are right

File: build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts (hunk 1 above)
Comment (evant) [reply build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts:24]: seems good on macos too

**Review by evant — APPROVED** [review]

**Pull request merged**